### PR TITLE
Clarify current VRF tie breaking implementation

### PR DESCRIPTION
**Pull request by amesgen**
# Description

Update/correct potentially misleading names of VRF-related functions (in particular `csvLeaderVRF`), and provide context for why a particular VRF value is used for tie breaking.   

# Checklist

- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Commits have useful messages
    - [ ] New tests are added if needed and existing tests are updated
    - [ ] If this branch changes Consensus and has any consequences for downstream repositories or end users, said changes must be documented in [`interface-CHANGELOG.md`](../ouroboros-consensus/docs/interface-CHANGELOG.md)
    - [ ] If this branch changes Network and has any consequences for downstream repositories or end users, said changes must be documented in [`interface-CHANGELOG.md`](../docs/interface-CHANGELOG.md)
    - [ ] If serialization changes, user-facing consequences (e.g. replay from genesis) are confirmed to be intentional.
- Pull Request
    - [x] Self-reviewed the diff
    - [x] Useful pull request description at least containing the following information:
      - What does this PR change?
      - Why these changes were needed?
      - How does this affect downstream repositories and/or end-users?
      - Which ticket does this PR close (if any)? If it does, is it [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)?
    - [x] Reviewer requested

### CODE REVIEW

**Review by Jasagredo — APPROVED** [review]
Looks reasonable to me

File: ouroboros-consensus-protocol/src/Ouroboros/Consensus/Protocol/Praos/Common.hs [hunk 1]
```diff
@@ -49,13 +49,14 @@ newtype MaxMajorProtVer = MaxMajorProtVer
...
--- 3. By the leader value of the chain tip, with lower values preferred.
+-- 3. By a VRF value from the chain tip, with lower values preferred. See
+--    @pTieBreakVRFValue@ for which one is used.
 data PraosChainSelectView c = PraosChainSelectView
   { csvChainLength :: BlockNo,
     csvSlotNo      :: SlotNo,
     csvIssuer      :: SL.VKey 'SL.BlockIssuer c,
     csvIssueNo     :: Word64,
-    csvLeaderVRF   :: VRF.OutputVRF (VRF c)
+    csvTieBreakVRF :: VRF.OutputVRF (VRF c)
```
Comment (dnadales) [inline ouroboros-consensus-protocol/src/Ouroboros/Consensus/Protocol/Praos/Common.hs:59]: Why do we think this is a better name than the previous one? Aren't in any case `csvChainLength` and `csvSlotNo` tie breakers? What about `csvChainTipVRF`?

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Protocol/Praos.hs [hunk 2]
```diff
@@ -138,7 +138,13 @@ instance PraosCrypto c => ProtocolHeaderSupportsProtocol (Praos c) where
       }
   pHeaderIssuer = hbVk . headerBody
   pHeaderIssueNo = SL.ocertN . hbOCert . headerBody
-  pHeaderVRFValue = certifiedOutput . hbVrfRes . headerBody
+
+  -- This is the "unified" VRF value, prior to range extension which yields e.g.
+  -- the leader VRF value used for slot election.
+  --
+  -- In the future, we might want to use a dedicated range-extended VRF value
```
Comment (dnadales) [inline ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Protocol/Praos.hs:145]: In the future or in `TPraos`, or "in future protocols"?

**Review by dnadales — APPROVED** [review]
I left one small question. Thanks for adding those comments to `pTieBreakVRFValue`.

File: ouroboros-consensus-protocol/src/Ouroboros/Consensus/Protocol/Praos/Common.hs (hunk 1 above)
Comment (amesgen) [reply ouroboros-consensus-protocol/src/Ouroboros/Consensus/Protocol/Praos/Common.hs:59]: In `TPraos`, `csvLeaderVRF` is the correct name, as we actually use the leader VRF value for tie breaking. But in `Praos` (which uses the same `SelectView`, i.e. this one), we no longer use the leader VRF, so this name is incorrect in that situation.

Hence, I chose a name indicating the purpose of the VRF in the context of chain selection, as it is the final tie breaker, compared to e.g. `csvChainLength` which causes the ties in the first place. (Side remark: `csvSlotNo` is not used for anything, we could remove it.)

But we don't have to stress its role as a tie breaker in the field name, e.g. `csvChainTipVRF` also sounds reasonable, as it also (intentionally!) does not mention which exact VRF from the header is used.

Comment (dnadales) [reply ouroboros-consensus-protocol/src/Ouroboros/Consensus/Protocol/Praos/Common.hs:59]: > Hence, I chose a name indicating the purpose of the VRF in the context of chain selection, as it is the final tie breaker,

I see, that makes sense. I'm ok with that name. Thanks!

**Comment by amesgen** [comment]
bors merge

**Comment by nfrisby** [comment]
I'm not 100% sure, but I suspect the rename of `csvTieBreakVRF` might require a (small) integration in eg `cardano-node` and so should be mentioned in the `interface-CHANGELOG.md`.

Did you check for that sort of thing already, @amesgen ?

**Comment by amesgen** [comment]
Grepping for `csvLeaderVRF` does not reveal anything in node (and also not in a few other IOG repos I have checked out). From my intuition, a `SelectView` should be a purely Consensus-internal thing, so I would actually be surprised/consider it tech debt if it would show up in consumers of the Cardano protocol. Where you thinking about a specific way it might be used in e.g. the node?

**Pull request merged**